.github/workflows/docs.yml

```diff
@@ -0,0 +1,33 @@
+name: Build and Deploy docs
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+    docs:
+      name: build and deploy docs
+      runs-on: ubuntu-latest
+
+      steps:      
+        - name: Checkout code
+          uses: actions/checkout@v2
+        - name: Set system to non-interactive mode
+          run: export DEBIAN_FRONTEND=noninteractive
+        - name: install dependencies
+          run: |
+            pip install sphinx
+            pip install sphinx-rtd-theme
+        - name: build docs
+          run: |
+            cd doc/sphinx
+            make html
+        - name: Deploy
+          uses: peaceiris/actions-gh-pages@v3.7.3
```

Comment:
I don't know how I feel about using this 3rd-party github action. But it is open source, and it's running only on github. I skimmed the typescript but I didn't follow most of it. If people would feel more comfortable with it, I can write the procedure manually.